tests/Command/BuyItemCommandTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Command;
+
+use PHPUnit\Framework\TestCase;
+use VendingMachine\Command\BuyItemCommand;
+use VendingMachine\Coordinator\PaymentCoordinator;
+use VendingMachine\Item\ItemB;
+use VendingMachine\Model\Dollar;
+use VendingMachine\Model\MoneyCollection;
+use VendingMachine\Repository\InMemoryItemRepository;
+use VendingMachine\Request\BuyItemRequest;
+use VendingMachine\Response\ConsoleResponse;
+
+class BuyItemCommandTest extends TestCase
+{
+    public function testShouldReturnBoughtBItem(): void
+    {
+        $itemRepository = new InMemoryItemRepository();
+        $itemRepository->add(new ItemB());
+
+        $buyItemCommand = new BuyItemCommand($itemRepository, new PaymentCoordinator(), new ConsoleResponse());
+
+        $result = $buyItemCommand(
+            new BuyItemRequest('B', new MoneyCollection([Dollar::create()]))
+        );
+
+        self::assertInstanceOf(ConsoleResponse::class, $result);
+    }
+
+    public function testShouldCalculateRestAndReturnResponseWithCorrectAmount(): void
+    {
+        $itemRepository = new InMemoryItemRepository();
+        $itemRepository->add(new ItemB());
+
+        $buyItemCommand = new BuyItemCommand($itemRepository, new PaymentCoordinator(), new ConsoleResponse());
+
+        $result = $buyItemCommand(
+            new BuyItemRequest('B', new MoneyCollection([Dollar::create(),]))
+        );
+
+        self::assertEquals(0, $result->rest()->count());
+    }
```

Comment:
You wrote only happy path cases. Where are cases to support errors like not enough money?